File: python/VendedorInterface.py
# coding: utf-8

from abc import ABC,abstractmethod
from EstoqueInterface import EstoqueInterface
from Vendedor import Vendedor
from Utils import Utils

class VendedorInterface(ABC):

    @classmethod
    def menu(cls, vendedores, montarias):
        while True:
            op = Utils.inputER("\n╔════════ MENU DE VENDEDORES ════════╗\n╠0 Voltar\n╠1 Adicionar Vendedor\n╠2 Remover Vendedor\n╠3 Visualizar Vendedores\n╚4 Estoque Vendedor\n⟶ ", int, "Entrada inválida, por favor escolha uma das opções.")

            if op == 0:
                return

            elif op == 1:
                cls.adicionaVendedor(vendedores)

            elif op == 2:
                cls.removeVendedor(vendedores)

            elif op == 3:
                cls.visualizaVendedores(vendedores)

            elif op == 4:
                vend = cls.selecionaVendedor(vendedores)
                if (vend != None):
                    EstoqueInterface.menu(vend.getEstoque(), vend.getNome(), montarias)

            else:
                print("Não existe esta opção, por favor digite novamente.")

    @staticmethod
    def criaVendedor():
        print("\nAdicionando Vendedor...")
        return Vendedor(input("Nome: "), input("Raça: "), Utils.inputER("Idade: ", int, "Entrada inválida, insira um número inteiro para a idade."), input("Descrição: "))

    @classmethod
    def adicionaVendedor(cls,vendedores):
        vendedores.append(cls.criaVendedor())

    @staticmethod
    def removeVendedor(vendedores):

        vendId = Utils.inputER("\nQual vendedor deseja remover? [Digite 0 para voltar] ", int, "Entrada inválida, insira o ID do vendedor que deseja remover.")

        if vendId == 0:
            return

        for v in vendedores:
            if v.getId() == vendId:
                vendedores.remove(v)
                print("Vendedor removido.")
                return

        print("Não há nenhum vendedor com o ID " + str(vendId) + " na base de dados.")

    @staticmethod
    def selecionaVendedor(vendedores):

        vendId = Utils.inputER("\nSelecione um vendedor para acessar o estoque [Digite 0 para voltar] ", int, "Entrada inválida, insira o ID do vendedor que deseja acessar o estoque.")

        if vendId == 0:
            return None

        for v in vendedores:
            if v.getId() == vendId:
                return v

        print("Nao há nenhum vendedor com o ID " + str(vendId) + " na base de dados")

    @staticmethod
    def visualizaVendedores(vendedores):

        if len(vendedores) == 0:
            print("\nNão há nenhum vendedor cadastrado na base de dados.")
            return

        print("")

        for v in vendedores:
            print("[{}] {}, {}, {} anos".format(v.getId(), v.getNome(), v.getRaca(), v.getIdade()))

File: java/src/main/java/venda_montaria/tp1/VendedorInterface.java
package venda_montaria.tp1;

import java.util.ArrayList;
import java.util.InputMismatchException;
import java.util.Scanner;

public abstract class VendedorInterface {

	public static void menu(ArrayList<Vendedor> vendedores, ArrayList<Montaria> montarias, Scanner sc) {

		while (true) {

//			visualizaVendedores(vendedores);

			int op;
			while (true) {
				System.out.println("\n╔════════ MENU DE VENDEDORES ════════╗");
				System.out.printf("╠0 Voltar\n╠1 Adicionar Vendedor\n╠2 Remover Vendedor\n╠3 Visualizar Vendedores\n╚4 Estoque Vendedor\n⟶ ");

				try {
					op = sc.nextInt();
					break;
				}
				catch(InputMismatchException e) {
					sc.nextLine();
					System.out.printf("Entrada inválida, por favor escolha uma das opções.\n ");
				}
			}

			
			switch (op) {
			case 0:
				return;
			case 1:
				sc.nextLine();
				adicionaVendedor(vendedores, sc);
				break;
			case 2:
				removeVendedor(vendedores, sc);
				break;
			case 3:
				visualizaVendedores(vendedores);
				break;
			case 4:
				Vendedor vend = selecionaVendedor(vendedores, sc);
				if (vend != null)
					EstoqueInterface.menu(vend.getEstoque(), vend.getNome(), montarias, sc);
				break;
			default:
				System.out.println("\nNão existe esta opção, por favor digite novamente.");

			}
		}
	}
	
	
	public static Vendedor criaVendedor(Scanner sc) {

		System.out.println("\nAdicionando vendedor...");
		System.out.printf("Nome: ");
		String nome = sc.nextLine();
		System.out.printf("Raça: ");
		String raca = sc.nextLine();
		int idade;
		while (true) {
			System.out.printf("Idade: ");
			try {
				idade = sc.nextInt();
				break;
			}
			catch(InputMismatchException e) {
				sc.nextLine();
				System.out.println("Entrada inválida, insira um número inteiro para a idade.");
			}
		}
		System.out.printf("Descrição: ");
		sc.nextLine();
		String descricao = sc.nextLine();

		return new Vendedor(nome, raca, idade, descricao);
	}

	
	public static void adicionaVendedor(ArrayList<Vendedor> vendedores, Scanner sc) {
		Vendedor novo_vendedor = criaVendedor(sc);
		vendedores.add(novo_vendedor);
	}


	public static void removeVendedor(ArrayList<Vendedor> vendedores, Scanner sc) {

		
		int vendId;
		while(true) {
		while (true) {
			System.out.printf("\nQual vendedor deseja remover? [Digite 0 para voltar] ");
			try {
				vendId = sc.nextInt();
				break;
			}
			catch(InputMismatchException e) {
				sc.nextLine();
				System.out.println("Entrada inválida, insira o ID do vendedor que deseja remover.");
			}
		}

		
		if (vendId == 0)
			return;

		for (Vendedor v : vendedores) {
			if (v.getId() == vendId) {
				vendedores.remove(v);
				System.out.println("Vendedor removido. ");
				return;
			}
		}
		System.out.printf("Não há nenhum vendedor com o ID " + vendId + " na base de dados.");
		}
	}

	public static Vendedor selecionaVendedor(ArrayList<Vendedor> vendedores, Scanner sc) {
		
		
		int vendId;
		while (true) {
				System.out.printf("Selecione um vendedor para acessar o estoque [Digite 0 para voltar]: ");
				try {
				vendId = sc.nextInt();
				break;
			}
			catch(InputMismatchException e) {
				sc.nextLine();
				System.out.println("Entrada inválida, insira o ID do vendedor que deseja acessar o estoque.");
			}
		}

		if (vendId == 0)
			return null;
		
		for (Vendedor v : vendedores)
			if (v.getId() == vendId)
				return v;

		System.out.println("Não há nenhum vendedor com o ID " + vendId + " na base de dados.\n");
		return null;
	}

	public static void visualizaVendedores(ArrayList<Vendedor> vendedores) {
		
		if (vendedores.size() == 0) {
			System.out.println("\nNão há nenhum vendedor cadastrado na base de dados.");
			return;
		}	
		
		System.out.println();
		
		for (Vendedor v : vendedores)
			System.out.printf("[%d] %s, %s, %d anos\n", v.getId(), v.getNome(), v.getRaca(), v.getIdade());
	}
}